### ISSUE TITLE
Project dead? Planned to make it open source for Safari?

### ISSUE THREAD

**Issue by whatthehomepod** [darkreader/darkreader#1205]

**Comment by drupadt** [comment]
Is this really dead?

**Comment by alexanderby** [comment]
Currently the amount of donations is too small, and the paid Safari version is the major source of revenue that makes this whole project live.

I'm going to change this by cleaning up the settings popover, the Donate button should become more visible. When the situation changes, the Safari version can become open-source.

If you mean that there were no updates for Safari in the past, I'm sorry for that. The problem is the Xcode IDE or Safari, don't know for sure. When developing the extension for Chrome, I can see the changes after 5 seconds, it reloads automatically. For Safari I have to restart the IDE multiple times after every minor change, it starts only in 10% of attempts. Debugging the Swift in Xcode is much more inconvenient comparing to JavaScript with Chrome Dev Tools. But recently I submitted an update to the App Store, it is currently under review. Usually it takes 2 days (compare to 1 hour for Chrome Web Store and 5 minutes for Firefox Add-ons).

**Comment by cryogenx** [comment]
@alexanderby so what you're saying is people using the safari extension are footing the bill for the development of the other extension and not receiving the same updates because "its too difficult"?

glad I looked in here before purchasing this add on

**Comment by anon-jpg** [comment]
@cyrogenx it’s just a complete ripoff, he started this as a hobby and wants people to pay like it’s a money teller machine. Jesus some developers these days. Some are ignorant af

**Comment by cryogenx** [comment]
I'm fine with him selling the product, so long as he's selling it to everyone, don't sell it to one set of users and give it to others... that's bs

**Comment by fluffynuts** [comment]
I dunno -- if it's orders of magnitude more difficult to develop for some platforms (and it appears it is -- and my experience is that apple stuff is harder to dev for, especially with webkit oddities and non-web-compliance), and especially if it's easier to monetize on those platforms, then sure, monetize wherever you can. 

Also, honestly: if you laid out $1500 for a (realistically) mid-range machine, you can afford $5 to make it transform the web for you. I bet there's plenty of software you've shelled out for on a Mac. I know that my short time with a Mac was not cheap: I had to buy plenty of add-on software to make the platform bearable. And I'm really not a fan of the "it cost me, so it should cost everyone" argument. That's how we end up with never-ending student debt and a host of similar crappy situations. And you may not realise how $5 actually costs another person a _lot_ more when their local currency is shit. But I digress.

I agree that the donation reminder / interface / whatever should be more obvious, especially if cash is needed to incentivise work. I have seen that I _can_ donate, but I've honestly forgotten by the time I'm back to "regular stuff". Perhaps if there was a PayPal button on the DarkReader settings overlay? Or a once-a-month-until-you-donate friendly reminder?

@cryogenx have you considered getting someone you trust to help out? I see that you do accept PRs (according to documentation), but there are (currently) 34 outstanding, which suggests that perhaps you need someone to donate time to assist with PR work.

Of course, that has to be someone you know and trust, so, as much as I can volunteer my time, it really doesn't count. I think DarkReader is really useful -- even though I've had to downgrade my current plugin to avoid strange site behavior. This project does something I'd started tinkering with in TamperMonkey -- and does it way better than I was getting to, so kudos. It really feels like the mission here is to get a plan:
- how to deal with PRs (not all have to be accepted!)
- how to divvy up the time for issue triangulation -- the current 445 count shows that (a) there are a lot of people who like this work and (b) there's stuff to be done. Probably quite a few of those issues are duplicates or at least related.

I can only imagine how the PR and issue backlog overwhelms -- when I had like 1 or 2 PRs against an open project, in addition to my daily workload, I was already feeling like it's getting a bit much!

I'm open to help (because it would honestly benefit me: DarkReader has made my web experience way better; also it would benefit a lot of people for whom the web has accessibility issues). You're welcome to vet against my code (eg synchronous-promise) or set me a "trial task", if you like. I'll honestly say that I don't have unlimited time though, so I may not be enough (:

**Comment by FranklinYu** [comment]
@fluffynuts You are replying to @cryogenx but actually the maintainer is @alexanderby.

**Comment by alexanderby** [comment]
@cryogenx The code, that makes webpages dark is 99% the same for both platforms. The difference is in user interface and background work (settings management and styles/images loading). I've started redesigning the settings page, so it will be easier to sync some features across the platforms. The situation that million of free users pays as much a thousand of paid users is a paradox and could be an interesting topic for research.

@anon-jpg Turning a hobby into a job, this is what everyone should strive for.

@fluffynuts My opinion is that price should correlate with [GDP per capita](https://en.wikipedia.org/wiki/List_of_countries_by_GDP_(nominal)_per_capita), but there's no such option in Mac App Store. In most of the countries it is even higher than in the US, because the price includes VAT.

I periodically merge the Pull Requests (more than 600 at the moment). Those left are usually invalid and their authors don't respond to my comments. Some require the changes in the app.

The number of issues is huge, but you should consider that the app has more than 1,400,000 users now. I'm trying to fix issues for popular websites. For other websites there is a workaround like Filter or Static mode. And also there are some feature requests that I'm going to implement. I agree that it turned into a mess and needs to be cleaned-up.

Overall I agree that there was no valuable progress this summer. There are several reasons. First of all I have some kind of a parental leave. Secondly the bureaucracy in Belarus is huge, e.g. I cannot use the donations for reasons different from healthcare or cultural; every year I have to request the IRS certificate proving that Apple Inc. is a US company and every 3 months report how much was their commission; there are many other crazy legal issues that eat up much time and there's no lawyer who could help me with this. Thirdly I'm under hiring process for some large big tech companies, so I have to prepare for and pass the interviews, solve some homework and spend much time studying migration laws in different countries.

I will fix some critical issues and release the update for Safari in the meantime. Stay tuned.

**Comment by anon-jpg** [comment]
You are right about that, but then you have to support it, maintain it, etc.

**Comment by fluffynuts** [comment]
@alexanderby thanks for all your hard work. At no point did I want to convey that I think that not enough is being done -- quite the opposite. If there's a way I can help out, please let me know.

**Comment by cryogenx** [comment]
If that the business model he wants to pursue that’s fine, and I get it.

I’d be more apt to spend my hard earned money on it if there was a trial version for Mac.. been burned to many times by pay only software that promises the world and doesn’t deliver

**Comment by ibash** [comment]
FWIW I paid for the safari version, and it works well for almost all cases. Consider how much time you spend in the browser at night and the safari version is ridiculously cheap. Especially considering how much better it makes the user experience, and that safari is much more efficient than chrome.

It's not a subscription, so you're paying for the software as it exists today, you are not paying for maintenance.

**Comment by alexanderby** [comment]
@ibash For some regions the app can be expensive, especially when the VAT is included. Unfortunately the App Store provides fixed tiers only. I think the most fare would be the price that correlates with a GDP per capita in a region. The reason for one time payment is its simplicity from the development point of view. At the same time refunds are available.

### An update for the issue

Unfortunately donations are still low (even with 2,000,000 users), and I found no way to legally use them in my country. So this revenue from Safari is very helpful, and I'm not going to make the app open source in the meantime, though the work on more code unification continues.

I've provided some fixes for the most popular issues and keep working on the app redesign. Unfortunately in November and December I was busy job interviews in Amazon and some legal issues. Working with the App Store is also barely legal in my country and different problems arise. But I decided to concentrate on coding for the first quarter of 2020. We will see what happens next.

So, a big update is planned for the end of this month.

**Comment by anon-jpg** [comment]
Spare that. iOS 13’s Dark Mode will be extended more.

Verstuurd vanaf mijn iPhone

**Comment by Gusted** [comment]
#2927 :)

**Comment by fluffypony** [comment]
Not to necro this thread, but frankly $5 as a once-off cost is totally fine.

That said: if I may be so bold, @alexanderby, what if for the next major Safari release you brought it to feature parity with the open-source one, and then charged an annual subscription of $0.99 for it? That way the revenue is residual instead of needing to constantly acquire new users. You can also deprecate the old version (ie. remove it from the store) after pushing out the last update that annoys people to update to the new version, and then the old version will continue working for some time at least.
